cpp/tests/strings/contains_tests.cpp

```diff
@@ -354,6 +355,57 @@ TEST_F(StringsContainsTests, CountTest)
   }
 }
 
+TEST_F(StringsContainsTests, FixedQuantifier)
+{
+  auto input = cudf::test::strings_column_wrapper({"a", "aa", "aaa", "aaaa", "aaaaa", "aaaaaa"});
+  auto sv    = cudf::strings_column_view(input);
+
+  {
+    // exact match
+    auto results = cudf::strings::count_re(sv, "a{3}");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 0, 1, 1, 1, 2});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // range match (greedy quantifier)
+    auto results = cudf::strings::count_re(sv, "a{3,5}");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 0, 1, 1, 1, 1});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // minimum match (greedy quantifier)
+    auto results = cudf::strings::count_re(sv, "a{2,}");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 1, 1, 1, 1, 1});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // range match (lazy quantifier)
+    auto results = cudf::strings::count_re(sv, "a{2,4}?");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 1, 1, 2, 2, 3});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // minimum match (lazy quantifier)
+    auto results = cudf::strings::count_re(sv, "a{1,}?");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({1, 2, 3, 4, 5, 6});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // zero match
+    auto results = cudf::strings::count_re(sv, "aaaa{0}");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 0, 1, 1, 1, 2});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    // poorly formed
+    auto results = cudf::strings::count_re(sv, "aaaa{n,m}");
+    cudf::test::fixed_width_column_wrapper<int32_t> expected({0, 0, 0, 0, 0, 0});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+    results = cudf::strings::count_re(sv, "aaaa{1234,5678}");
```

Comment:
Oh right, it just treats `a{100000}` as a literal and does not recognize the `{}` as special characters in this case. This is what it was doing before in cases like `a{111` or `a{bcd}`.  It just would be as if someone escaped it like `a\{bcd\}` 
The value must have a max since we cannot store arbitrary large numbers. I just set it to something reasonable instead of int-max. I was not looking to add new behavior (i.e. throw exception) but now I'm reconsidering that. What do you think?

Reply:
I think that if someone has a non-numerical value in curly braces like `a{bcd}` it is reasonable to assume that they wanted literal curly braces. On the other hand, having curly braces behave differently depending on the size of the number in braces violates the principle of least surprise, so I would strongly advocate for raising an exception there. I think it's completely reasonable to have a maximum value, just not to silently treat the regex as a literal once that threshold is exceeded.